DEVELOPING.md

```diff
@@ -1,14 +1,97 @@
+<!-- TOC -->
+
+- [Requirements](#requirements)
+- [Setting up your development environment](#setting-up-your-development-environment)
+- [Building the project from source](#building-the-project-from-source)
+- [Development Tasks](#development-tasks)
+        - [Basic Verification](#basic-verification)
+        - [Docsite](#docsite)
+    - [Details](#details)
+        - [Coding Style](#coding-style)
+        - [Lint](#lint)
+        - [Type checking](#type-checking)
+        - [Unit tests](#unit-tests)
+- [Advanced Updating submodules](#advanced-updating-submodules)
+- [Cython and C++](#cython-and-c)
+- [Making a Release](#making-a-release)
+
+<!-- /TOC -->
+
 # Requirements
-* Python 3.8+
-* Poetry (`curl -sSL https://install.python-poetry.org | python - --version=1.2.2`)
 
-For the other requirements, inspect the ``pyproject.toml`` file. If you are updated the dependencies, please run `poetry update` to update the
+* Python 3.9+
+* numpy
+* scipy
+* scikit-learn>=1.3.1
+
+For the other requirements, inspect the ``pyproject.toml`` file.
+
+# Setting up your development environment
+
+We recommend using miniconda, as python virtual environments may not setup properly compilers necessary for our compiled code. 
+
+<!-- Setup a conda env -->
+
+    conda create -n sktree python=3.9 
```

Comment:
Don't think we need to specify python version. We could maybe link to how to create conda envirnoments before the code